File: 2020/a.py
#!/usr/bin/env python3

# Problem A - Count Down Up 2020

L2020 = '2 0 2 0'.split()

def solve(ds):
    return list(ds[i : i + 4] == L2020
                for i in range(len(ds))).count(True)

while True:
    n = int(input())
    if n == 0:
        break
    ds = input().split()
    assert len(ds) == n
    print(solve(ds))

File: 2020/a.hs
#!/usr/bin/env runghc

-- ACM-ICPC 2020 Domestic Qualifier
-- Problem A: Count Down Up 2020

import Control.Monad (when)
import Control.Applicative ((<$>))

main :: IO ()
main = do
    n <- read <$> getLine :: IO Int
    when (n /= 0) $ do
        ds <- map read . words <$> getLine :: IO [Int]
        print $ solve ds
        main

solve :: [Int] -> Int
solve (2:0:2:0:ds) = 1 + solve (2:0:ds)
solve (a:ds) = solve ds
solve _ = 0
